builtin/providers/google/resource_google_service_account_test.go

```diff
@@ -0,0 +1,51 @@
+package google
+
+import (
+	"fmt"
+	"testing"
+
+	"github.com/hashicorp/terraform/helper/resource"
+	"github.com/hashicorp/terraform/terraform"
+)
+
+var (
+	accountId   = "tf-test"
+	displayName = "Terraform Test"
+)
+
+// Test that a service account resource can be created and destroyed
+func TestAccGoogleServiceAccount_basic(t *testing.T) {
+	resource.Test(t, resource.TestCase{
+		PreCheck:  func() { testAccPreCheck(t) },
+		Providers: testAccProviders,
+		Steps: []resource.TestStep{
+			resource.TestStep{
+				Config: fmt.Sprintf(testAccGoogleServiceAccount_basic, accountId, displayName),
+				Check: resource.ComposeTestCheckFunc(
+					testAccCheckGoogleServiceAccountExists("google_service_account.acceptance"),
+				),
+			},
+		},
+	})
+}
+
+func testAccCheckGoogleServiceAccountExists(r string) resource.TestCheckFunc {
+	return func(s *terraform.State) error {
+		rs, ok := s.RootModule().Resources[r]
+		if !ok {
+			return fmt.Errorf("Not found: %s", r)
+		}
+
+		if rs.Primary.ID == "" {
+			return fmt.Errorf("No ID is set")
+		}
+
+		return nil
+	}
+}
+
+var testAccGoogleServiceAccount_basic = `
+resource "google_service_account" "acceptance" {
+    account_id = "%v"
```

Comment:
Instead of passing static variables, would it not be preferable to have a test that passes a random character string, similar to the other google provider tests (example: https://github.com/hashicorp/terraform/blob/45f441fdb4c0ec05d2ac70d76d07dd7c83930b98/builtin/providers/google/resource_compute_route_test.go#L104-L110)

This would ensure a test wouldn't falsely pass because a resource wasn't cleaned up matching the static name. Or is there some reason I'm missing that wouldn't work for this resource type?